File: footnest_app/lib/models/standing/standing.dart
import '/models/team/team.dart';

class Standing {

  final Team team;
  final int position;
  final int played;
  final int wins;
  final int draws;
  final int losses;
  final int goalsFor;
  final int goalsAgainst;
  final int points;
  final double totalXg;

  Standing({
    required this.team,
    required this.position,
    required this.played,
    required this.wins,
    required this.draws,
    required this.losses,
    required this.goalsFor,
    required this.goalsAgainst,
    required this.points,
    required this.totalXg,
  });

  factory Standing.fromJson(Map<String,dynamic> json){
    return Standing(
      team: Team.fromJson(json['team']),
      position: json['position'],
      played: json['played'],
      wins: json['wins'],
      draws: json['draws'],
      losses: json['losses'],
      goalsFor: json['goalsFor'],
      goalsAgainst:
          json['goalsAgainst'],
      points: json['points'],
      totalXg:
          json['totalXg']?.toDouble() ?? 0,
    );
  }

}

File: footnest-backend/src/main/java/com/footnest/footnest_backend/entity/Standing.java
package com.footnest.footnest_backend.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(
    name = "standings",
    uniqueConstraints = {
        @UniqueConstraint(
            columnNames = {
                "competition_season_id",
                "team_id"
            }
        )
    }
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Standing {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "competition_season_id", nullable = false)
    private CompetitionSeason competitionSeason;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "team_id", nullable = false)
    private Team team;


    private Integer played = 0;

    private Integer wins = 0;

    private Integer draws = 0;

    private Integer losses = 0;


    private Integer goalsFor = 0;

    private Integer goalsAgainst = 0;


    private Integer points = 0;


    private Double totalXg = 0.0;
}